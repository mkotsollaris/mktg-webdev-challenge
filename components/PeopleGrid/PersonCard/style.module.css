.avatar {
  vertical-align: middle;
  width: 7rem;
  height: 7rem;
  border-radius: 50%;
  margin-bottom: 0.5rem;
}

.person {
  display: grid;
  border: 1px solid lightgray;
  padding: 1rem;
  text-align: center;
  &:nth-child(1) {
    max-width: 20rem;
  }

  & .centered {
    align-self: center;
    justify-self: center;
  }

  & svg {
    width: 10rem;
  }
}

.light {
  color: darkslategray;
  font-size: 0.8rem;
}
