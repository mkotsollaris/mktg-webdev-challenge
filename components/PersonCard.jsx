import style from './style.module.css'
import logo from '../public/img/avatar.svg';

const PersonCard = ({ url, name, title, department }) => {
    return <div className={style.person}>
        <div className={style.centered}>
            {url ? <img className={style.avatar} src={url} /> : <img src={logo} className={style.avatar} alt="logo" />}
        </div>
        <div className={style.name}>
            {name}
        </div>
        <div>
            {title}
        </div>
        <div>
            {department}
        </div>
    </div>
}

export default PersonCard