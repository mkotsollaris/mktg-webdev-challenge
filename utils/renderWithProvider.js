import { render } from '@testing-library/react'
import AppContext from '../context/AppContext'

function renderWithProvider(props, Component) {
  return render(
    <AppContext.Provider value={props}>
      <Component />
    </AppContext.Provider>
  )
}

export default renderWithProvider
