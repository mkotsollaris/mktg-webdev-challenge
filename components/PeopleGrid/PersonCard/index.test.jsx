import PersonCard from '.'
import renderWithProvider from '../../../utils/renderWithProvider'

it('renders', () => {
  const { asFragment } = renderWithProvider({}, PersonCard)
  expect(asFragment()).toMatchSnapshot()
})
