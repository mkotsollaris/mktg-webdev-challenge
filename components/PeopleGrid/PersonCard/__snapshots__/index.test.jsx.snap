// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
    <div>
      <img
        alt="avatar"
        src="[object Object]"
      />
    </div>
    <div />
    <div />
    <div />
  </div>
</DocumentFragment>
`;
